TaskHistory.Impl/Users/UserRepo.cs

```diff
@@ -3,51 +3,65 @@
 using TaskHistory.Impl.MySql;
 using MySql.Data.MySqlClient;
 using System.Data;
+using System.Configuration;
 
 namespace TaskHistoryImpl.Users
 {
-	public class UserRepo : AbstractMySqlRepo, IUserRepo
+	public class UserRepo : IUserRepo
 	{
 		private readonly UserFactory _userFactory;
 
 		public IUser ValidateUsernameAndPassword (string username, string password)
 		{
-			var command = _mySqlCommandFactory.CreateMySqlCommand ("User_Validate");
-			command.Parameters.Add (new MySqlParameter ("pUsername", username));
-			command.Parameters.Add (new MySqlParameter ("pPassword", password));
+			using (var connection = new MySqlConnection (ConfigurationManager.AppSettings ["MySqlConnection"]))
+			using (var command = new MySqlCommand ("User_Validate", connection)) 
+			{
+				command.CommandType = CommandType.StoredProcedure;
 
-			MySqlDataReader reader = command.ExecuteReader (CommandBehavior.CloseConnection);
+				command.Parameters.Add (new MySqlParameter ("pUsername", username));
+				command.Parameters.Add (new MySqlParameter ("pPassword", password));
+				command.Connection.Open ();
 
-			IUser user = null;
+				using (var reader = command.ExecuteReader (CommandBehavior.CloseConnection)) 
+				{
+					IUser user = null;
 
-			if (reader.Read ()) 
-			{
-				user = CreateUserFromReader (reader);
-			}
+					if (reader.Read ()) 
+					{
+						user = CreateUserFromReader (reader);
+					}
 
-			return user;
+					return user;
+				}
+			}
 		}
 
 		public IUser RegisterUser (string username, string password, string firstName, string lastName, string email)
```

Comment:
we can wrap username and password as one object.

firstName, lastName, and email as a dto object.